src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,154 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> addedNode = new Node<>(null, value, null);
+
+        if (isEmpty()) {
+            head = addedNode;
+        } else {
+            addedNode.prev = tail;
+            tail.next = addedNode;
+        }
+
+        tail = addedNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+
+        if (index == 0) {
+            head = new Node<>(null, value, head);
+            head.next.prev = head;
+        } else {
+            isInBounds(index);
+            Node<T> nodeAtIndex = iterateToNode(index);
```

Comment:
This is code repeating. You always check the index before getting node by index. What you should do is move the check logic INSIDE the get method.